### PR TITLE
[Sofa.GUI] Clean CMake variables

### LINKED ISSUE #2972: Commit 1118d9de breaks VSync disabling
### Problem

**Description**
Commit 1118d9de breaks the ability to switch VSynch to false. The simulation cannot go faster than the refreshing rate of the screen even if the nividia-setting VSync parameter is set to off. 

**Steps to reproduce**
Just try to lanch runSofawith vsync set to false and the simulation will not go faster than your screen refreshing rate. 

**Expected behavior**
Should go faster

---------------------------------------------

### Environment

**Context**

- System: Ubuntu 22.04
- Version of SOFA: branch master anywhere after 1118d9de
- State: build in RelWithDebInfo 

**Command called**

sunSofa

```

**Env vars**

#################
--- sys.version ---
3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0]
--- PATH ---
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/paul/dev/build/sofa/bin:/home/paul/dev/sources/sofaProject/gitlab.inria.fr/mimesis/mimesiscript:/usr/local/cuda/bin
--- SOFA_ROOT ---
--- PYTHONPATH ---
--- sys.path ---
['', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/usr/local/lib/python3.10/dist-packages', '/usr/lib/python3/dist-packages']
#################


[CMakeCache.txt](https://github.com/sofa-framework/sofa/files/8738540/CMakeCache.txt)


Thank you for your report.

### PR DESCRIPTION
**Pull request by fredroy**
Clean all cmake var/options/names to follow the new guidelines and the new SOFA-NG module names
(fulfill some TODO written long ago as well)

+ Fix #2972

Remark/question: should we keep compatibility with all the cmake variables/options and preproc defines?

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Review by alxbilger — COMMENTED** [review]
Does the CIs need to be updated with the new variable names?

**Comment by fredroy** [comment]
> Does the CIs need to be updated with the new variable names?

I think the CI does not really set any GUI/Qt cmake variables.
In any case, as there is no forwarding from the new var to the old deprecated var, it should be yes.

File: Sofa/GUI/CMakeLists.txt [hunk 1]
```diff
@@ -4,12 +4,12 @@ add_subdirectory(Component)
...
-if(SOFAGUI_QT)
+option(SOFA_GUI_ENABLE_QT "Compile the Qt GUI" ON)
+if(SOFA_GUI_ENABLE_QT)
     add_subdirectory(Qt)
 endif()
 
-option(SOFAGUI_HEADLESS_RECORDER "Compile the headless recorder (Linux only)" OFF)
-if(SOFAGUI_HEADLESS_RECORDER)
+option(SOFA_GUI_ENABLE_HEADLESS_RECORDER "Compile the headless recorder (Linux only)" OFF)
+if(SOFA_GUI_ENABLE_HEADLESS_RECORDER)
```
Comment (guparan) [inline Sofa/GUI/CMakeLists.txt]:
```suggestion
option(SOFA_GUI_ENABLE_HEADLESSRECORDER "Compile the headless recorder (Linux only)" OFF)
if(SOFA_GUI_ENABLE_HEADLESSRECORDER)
```
Looks more consistent to me. May need to be propagated somewhere else...

**Review by guparan — CHANGES_REQUESTED** [review]

File: Sofa/GUI/CMakeLists.txt (hunk 1 above)
Comment (fredroy) [reply Sofa/GUI/CMakeLists.txt]: seems not (after grep'ping, it is the only two occurences)

**Review by guparan — APPROVED** [review]

**Comment by guparan** [comment]
This is breaking, right @fredroy ?

**Comment by fredroy** [comment]
> This is breaking, right @fredroy ?

Breaking in term of user configuration, yes ; as the cmake option names changed 🫤

**Comment by guparan** [comment]
And in term of SDK usage

**Pull request merged**